File: android/src/main/java/com/breezsdk/BreezSDKListener.kt
package com.breezsdk

import breez_sdk.BreezEvent
import breez_sdk.EventListener
import com.facebook.react.modules.core.DeviceEventManagerModule.RCTDeviceEventEmitter

class BreezSDKListener(private val emitter: RCTDeviceEventEmitter): EventListener {
    companion object {
        var emitterName = "breezSdkEvent"
    }

    override fun onEvent(e: BreezEvent) {
        when (e) {
            is BreezEvent.InvoicePaid -> emitter.emit(emitterName, readableMapOf("type" to "invoicePaid", "data" to readableMapOf(e.details)))
            is BreezEvent.NewBlock -> emitter.emit(emitterName, readableMapOf("type" to "newBlock", "data" to e.block))
            is BreezEvent.PaymentFailed -> emitter.emit(emitterName, readableMapOf("type" to "paymentFailed", "data" to e.error))
            is BreezEvent.PaymentSucceed -> emitter.emit(emitterName, readableMapOf("type" to "paymentSucceed", "data" to readableMapOf(e.details)))
            is BreezEvent.Synced -> emitter.emit(emitterName, readableMapOf("type" to "synced"))
        }
    }
}

File: ios/BreezSDKListener.swift
import Foundation

class BreezSDKListener: NSObject, EventListener {
    var emitter: RCTEventEmitter
    
    static let emitterName: String = "breezSdkEvent"
    
    init(emitter: RCTEventEmitter) {
        self.emitter = emitter
    }
    
    func onEvent(e: BreezEvent) {
        switch(e) {
        case let .invoicePaid(details):
            self.emitter.sendEvent(withName: BreezSDKListener.emitterName,
                                   body: [
                                    "type": "invoicePaid",
                                    "data": BreezSDKMapper.dictionaryOf(invoicePaidDetails: details)
                                   ])
        case let .newBlock(block):
            self.emitter.sendEvent(withName: BreezSDKListener.emitterName,
                                   body: [
                                    "type": "newBlock",
                                    "data": block
                                   ])
        case let .paymentFailed(error):
            self.emitter.sendEvent(withName: BreezSDKListener.emitterName,
                                   body: [
                                    "type": "paymentFailed",
                                    "data": error
                                   ])
        case let .paymentSucceed(details):
            self.emitter.sendEvent(withName: BreezSDKListener.emitterName,
                                   body: [
                                    "type": "paymentSucceed",
                                    "data": BreezSDKMapper.dictionaryOf(payment: details)
                                   ])
       case .synced:
            self.emitter.sendEvent(withName: BreezSDKListener.emitterName,
                                   body: [
                                    "type": "synced"
                                   ])
        }
    }
}
